pg_shard.c

```diff
@@ -579,7 +579,7 @@ ExtractFirstDistributedTableId(Query *query)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
 
-		if (IsDistributedTable(rangeTableEntry->relid))
+		if (rangeTableEntry->rtekind == RTE_RELATION && IsDistributedTable(rangeTableEntry->relid))
```

Comment:
Note to @sumedhpathak: According to comments for `RangeTblEntry`, `relid` is only set for range tables whose kind is `RTE_RELATION`. The views rewrite to subqueries, which have a `relid` of zero, which we happily pass to `IsDistributedTable`. By adding this check we avoid that.

If the subqueries themselves contain distributed relations, that's OK, because the walker will have grabbed their entries and we'll come across them in this loop anyhow.